### ISSUE TITLE
Random CV folds shatter multi-site studies, breaking cohort identifiability and the predictive premise

### ISSUE THREAD

**Issue by bschilder** [bschilder/genomeOS#127]
Raised by @canderson50 while diagnosing #120, and it holds. Measured on the current MAP HbS corpus
(1,071 surveys, 419 studies, 143 of them multi-site):

| fold strategy | multi-site studies split across folds | share |
|---|---|---|
| spatial | 7 / 143 | **5%** |
| random | 135 / 143 | **94%** |

A 19x difference, and it lines up with #111's failure pattern: **3 of 5 random folds failed to
converge against 1 of 5 spatial**, and `cohort_sd` was the worst-mixing parameter in most of them.

## Why splitting a study breaks the fit

`cohort_id` is the contributing *study*, not the survey site — deliberately, because keying by site
would give one cohort level per observation, which is not a cohort effect at all (see the docstring
in `observations.sources.map_surveys`). The effect is identified by *within-study replication*: the
41-plus studies contributing more than one site are what make `cohort_sd` estimable.

A random split cuts those studies apart, so in each training set they appear as singletons. For a
singleton study, `cohort_sd * cohort_z` and the beta-binomial `concentration` describe the same
single residual and are not jointly identifiable — exactly the degeneracy #121 found in AFND, where
one study per population makes it global rather than incidental. k-means spatial folds keep a
study's sites together because sites from one study tend to be near each other, so the effect
survives.

**This means the spatial/random comparison in #109 is not a clean measurement of leakage.** The gap
was read as "random folds leak neighbours across the split, so their skill is inflated" — that is
still true, but the two strategies also differ in whether the model they fit is identifiable at
all. Two effects, one number.

## It also breaks a premise in the predictive path

`SurfaceFit.predict_observation` (#110) integrates the cohort offset over `Normal(0, cohort_sd)`,
on the stated grounds that *"a held-out survey belongs to a cohort the model never saw"*. Under
random folds that is false for **94%** of multi-site studies: the model has seen that study, at
other sites, and conditioning on its fitted offset would be both available and more accurate.
Marginalising is right for a genuinely unseen cohort and needlessly wide for a seen one — so the
predictive intervals from random folds are conservative in a way that was not intended and is not
recorded.

## Cheap next steps, no GPU needed

- **Report the split count per strategy** in `CrossValidation`, so the number is visible next to
  the scores rather than needing a separate script.
- **Add a study-grouped fold strategy** (`GroupKFold` on `cohort_id`), which is the standard answer
  and isolates leakage from identifiability — it keeps studies intact while still holding out
  unseen ones.
- **Condition rather than marginalise** in `predict_observation` when the held-out survey's cohort
  appears in training, and say which was done per observation.

Credit to @canderson50 for spotting it from the code alone; the numbers above are the check they
suggested.

Related: #109, #110, #111, #121.

**Comment by bschilder** [comment]
## This is not an isolated bug — it is the sixth appearance of one failure mode

Worth writing down, because it has now been rediscovered six times under six different names, and
each time the first reading was "the sampler needs more draws".

**Every one is the same shape: two parameters that can trade against each other, with nothing in
the data to say which is which.** Non-identifiability. The sampler's job is to explore a
distribution; when two knobs describe one quantity there is a ridge of equally good answers, so the
chains slide along it instead of converging.

| issue | the pair competing | status |
|---|---|---|
| **#83** binomial likelihood | no dispersion parameter, so the *spatial field* inflated to absorb survey-to-survey variance (amplitude 11.5, r_hat 2.61) | closed — beta-binomial |
| **#103** Nordic hotspot | same root: an over-inflated field extrapolating into empty space | closed — resolved by #83 + spherical kernel |
| **#116** G6PD | a very long correlation range vs the global `intercept` — a near-constant field *is* an intercept | open |
| **#122** lengthscale across 767 alleles | the same ridge, at scale: 19 of 20 HLA alleles "converged" to 1,036–4,111 km | open — data-derived prior shipped, unverified |
| **#121** AFND cohorts | `cohort_sd * cohort_z` vs beta-binomial `concentration` — one study per population, so every cohort is a singleton | open — term dropped when 1:1 |
| **#127** this issue | the *same* pair as #121, but the degeneracy is **created by how we split the data** rather than present in it | open |

## Why #121 and this are worth reading together

They are the identical degeneracy from opposite directions.

- In **#121** it is a property of the source. AFND publishes one study per population, so
  `cohort_id == population_id` permanently and no split can fix it. Dropping the term is right.
- Here it is **self-inflicted**. MAP has 143 multi-site studies and the effect is genuinely
  identifiable — until a random split shatters 94% of them and turns identified cohorts into
  singletons in training. Dropping the term would be wrong; fixing the split is the fix.

The same symptom therefore has opposite remedies, which is exactly why `ConvergenceError` naming
the offending parameter matters. `cohort_sd` starving means "this pair is fighting" — it does not
tell you whether the cause is the data or the fold strategy, and answering that needs the check in
this issue.

## What it contaminates downstream

- **#111** asked why random folds failed 3/5 against spatial's 1/5 and said it was "worth
  understanding rather than assuming it is noise". This is the answer, and it means some of those
  failures are not a sampling-budget problem at all — so raising `draws` may fix fewer of them
  than #120 hopes.
- **#109** reported 34% of apparent skill as autocorrelation leakage. Directionally right, but the
  two strategies also differ in whether the fitted model is identifiable, so that number is two
  effects under one label.
- **#110** integrates the cohort offset over its population because "a held-out survey belongs to a
  cohort the model never saw". False for 94% of multi-site studies under random folds.

## Suggested follow-up beyond this issue

Name it in `docs/overview.md` as a recognised failure mode, with the table above. Six independent
rediscoveries is a documentation gap, not six unlucky bugs — and a contributor who has read it once
will recognise the seventh in a single look at a diagnostic.

**Comment by bschilder** [comment]
Status, and the remaining scope is now unblocked.

## Done

**Proposals 1 and 2 landed in #131** (`4f5fc62`, authored by @canderson50):

- a `grouped` fold strategy assigning whole studies to folds, so no study is split and the cohort
  term stays identifiable in every training set;
- `studies_split_across_folds` computed and reported for **every** strategy, so the confound is
  visible rather than inferred.

**Measured on the full MAP HbS corpus** in #132, which is the confirmation this issue was asking
for:

| Fold strategy | Multi-site studies split | Folds converged |
|---|---|---|
| grouped | **0 / 143 (0%)** | 5/5 |
| spatial | 7 / 143 (5%) | 4/5 |
| random | **135 / 143 (94%)** | 4/5 |

The 94% figure this issue predicted is confirmed on real data, and the consequence is confirmed
too: the autocorrelation share of apparent skill is **37%**, measured spatial-versus-grouped. That
number was not computable before, because spatial-versus-random mixed spatial leakage with the loss
of identifiability described here.

## Remaining: proposal 3

**Conditioning rather than marginalising in `predict_observation` when the cohort was seen in
training.** On `main` today, `crossval.py` calls:

```python
replicated = fit.predict_observation(lat=lat, lon=lon, an=an)
```

with no `cohort_id`, so a study the model has already seen at other sites is integrated over rather
than conditioned on. That widens every predictive interval for no reason, and under random folds it
applies to 94% of multi-site studies.

#131 deliberately left this out because it rewrites the same predictive path as #124, which changes
when the cohort term is fitted at all. **#124 has now landed (`79d8c79`), so the collision is
gone.**

A working implementation exists on the unmerged local branch `fix/127-fold-grouping`:

```python
# Pass the cohort so a study the model saw at other sites is conditioned on rather than
# integrated over. Under random folds that is 94% of multi-site studies (#127), and
# marginalising them widens every interval for no reason.
replicated = fit.predict_observation(
    lat=lat, lon=lon, an=an, cohort_id=test["cohort_id"].to_numpy()
)
```

That branch predates #131 and was never opened as a PR, so it needs rebasing onto current `main`
and reconciling with `beta_cohort_applied` from #124 — when the cohort effect is not estimated,
there is no offset to condition on and the call must degrade to the current behaviour rather than
pass a cohort the fit never used.

## Why it matters more now, not less

#132 found 95% predictive coverage at **0.96 under grouped folds** and **0.74 under spatial**.
Coverage is the headline statistic, and marginalising over a cohort the model has already seen
inflates intervals — so the grouped 0.96 and the over-wide 50% coverage (0.62 against nominal 0.50)
are both measured under the *pessimistic* convention. Fixing this changes the calibration numbers
that #109 just closed on, which is a reason to do it deliberately and re-run rather than fold it
into unrelated work.

Keeping this issue open for proposal 3. Related: #109, #110, #124, #131, #132.

**Comment by bschilder** [comment]
Figure for this issue, added in #138 and generated by `scripts/plot_fold_strategies.py`:

![fold strategies](https://raw.githubusercontent.com/bschilder/genomeOS/d543baded12b90bd5e77d88784756b6b4d783b80/docs/figures/fold_strategies.png)

Bottom row is the substance — how many folds each multi-site study ends up spanning, where 1 means
intact:

| strategy | studies split | intact | shape of the damage |
|---|---|---|---|
| `random` | **135 / 143 (94%)** | 8 | 58 split two ways, 26 three, 27 four, **24 across all five folds** |
| `spatial` | 7 / 143 (5%) | 136 | 6 split two ways, 1 three |
| `grouped` | **0 / 143 (0%)** | 143 | none, by construction |

The detail worth adding to this issue's original argument: **24 studies have sites in all five
folds simultaneously.** For those there is no training split that retains any replication at all,
so the identifiability loss this issue describes is not a tendency, it is total in every run.

The maps also show why `grouped` and not `random` is the right baseline for measuring spatial
leakage. `grouped` looks nearly as scattered as `random` and leaks neighbours the same way, but
every study stays one colour — which is what makes the spatial-versus-grouped gap attributable to
autocorrelation alone, and how #132 got the 37% figure.
